index.js

```diff
@@ -102,6 +106,57 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  if (!fastify.hasRequestDecorator('pg')) {
+    fastify.decorateRequest('pg', null)
+  }
+
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.pg && routeOptions.pg.transact
+
+    if (useTransaction) {
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+
+        if (name) {
+          if (client[name]) {
+            return next(new Error(`pg client '${name}' is a reserved keyword`))
+          } else if (fastify.pg[name]) {
```

Comment:
this also doesn't make much sense, you must check against req.pg, not fastify.pg